README.md

```diff
@@ -130,6 +133,135 @@ For Facebook Messenger, the approach is the same, but the JSON details differ:
+          {
+            "label": "Location 2",
+            "value:" "Location 2"
+          },
+          {
+            "label": "Location 3",
+            "value:" "Location 3"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+**Slack**
+The generic response array is translated to a *single* message in Slack comprising of text alongwith attachment data. For the above example, the equivalent Slack message after translation is as follows:
+
+```json
+{
+  "text": "Here are your nearest stores.",
+  "attachments": [
+    {
+      "image_url": "http://...",
+      "pretext": "Some description for the image",
+      "title": "Image title"
+    },
+    {
+      "actions": [
+        {
+          "name": "Location 1",
+          "text": "Location 1",
+          "type": "button",
+          "value": "Location 1"
+        },
+        {
+          "name": "Location 2",
+          "text": "Location 2",
+          "type": "button",
+          "value": "Location 2"
+        },
+        {
+          "name": "Location 3",
+          "text": "Location 3",
+          "type": "button",
+          "value": "Location 3"
+        }
+      ],
+      "callback_id": "Click on one of the following",
+      "text": "Click on one of the following"
+    }
+  ],
+  "channel": "DXXXXXXXX",
+  "ts": "XXXXXXXXXX.XXXXXX"
+}
+}
+```
+**Facebook**
+The generic response array is translated to a *list of Facebook messages payloads* comprising of text along with attachment data. For the above example, the equivalent message list after translation is as follows:
+
+```json
+[
+  {
+    "text": "Here are your nearest stores."
+  },
+  {
+    "attachment": {
+      "type": "image",
+      "payload": {
+        "url": "http://..."
+      }
+    }
+  },
+  {
+    "text": "Click on one of the following",
+    "quick_replies": [
+      {
+        "content_type": "text",
+        "title": "Location 1",
+        "payload": "Location 1"
+      },
+      {
+        "content_type": "text",
+        "title": "Location 2",
+        "payload": "Location 2"
+      },
+      {
+        "content_type": "text",
+        "title": "Location 3",
+        "payload": "Location 3"
+      }
+    ]
+  }
+]
+```
+In case Facebook is the channel in question, the output of `normalize-conversation-for-<channel>` is sent to `multiple_post` which splits the array into individual Facebook messages and invokes a `sub-pipeline` comprising of `starter-code/post-normalize` and `facebook/post` functions. Each message is separately passed through this subpipeline. Hence, a multi-modal response from Conversation is broken down into individual messages and POSTed to Facebook messenger.
+
+Additionally, Facebook only allows upto 11 elements in a quick reply array. So, if the options list in the Conversation response has over 11 options, it's translated to the [Facebook generic template](https://developers.facebook.com/docs/messenger-platform/send-messages/template/generic) format with buttons in groups of three.
+
```

Comment:
commenting on the long block of text, this is a really nice write-up btw. I believe in the past we have refrained from showing detailed JSON, preferring to link to Conversation pages. Are response types going to be doced elsewhere? Maybe we can link and simplify? Or maybe that is a solution we can do later. Also wondering if we potentially want to push some of the facebook specific stuff to the facebook readme, I know we don't have an architecture section there now but maybe it is time to do so?

Reply:
Just to reiterate what I said in our call:

- I will check with Michelle Miller and make sure response types will be documented in the main Conversation docs.
- I don't think we need the last two JSON examples (the ones showing how the generic JSON is mapped to channel-specific JSON for both channels). We have already shown what channel-specific JSON looks like, and I think that's sufficient.
- I like the idea of keeping all of the architectural description in this readme for now, although if things get more complicated later (like if we add more channels) we might need to revisit that.